### ISSUE TITLE
[BUG]: USB - GunCon 2's calibration shots not always work as intended

### ISSUE THREAD

**Issue by IlDucci** [PCSX2/pcsx2#7618]
### Describe the Bug

I have tested some PAL GunCon 2 games and found out that the Calibration Shot does not work sometimes as it should:
 - Vampire Night PAL (SCES-50411) is not working at all. I'm not being capable of calibrating the gun, not with the Calibration Shot, not with the normal Trigger.
 - Dino Stalker PAL ENG (SLES-50930) always recognises the first Calibration Shot as a double press, making the game recognise only the horizontal movement. Pressing A/B and then recalibrating will make the process work correctly.

### Reproduction Steps

 - Go to the calibration screen of each game.
 - Try to calibrate the GunCon 2.

### Expected Behavior

- Vampire Night should let itself be calibrated. This is a major bug for this game.
- Dino Stalker should have no surprises on a first calibration pass. This bug can be workarounded by doing a second calibration.

### PCSX2 Revision

v1.7.3735

### Operating System

Windows 10 (64bit)

### CPU

i5-6600

### GPU

RTX 2060

### GS Settings

![imagen](https://user-images.githubusercontent.com/5641681/208320646-44017980-bb83-4bf0-b719-6a0b4510df0b.png)
![imagen](https://user-images.githubusercontent.com/5641681/208320656-a3f2dd72-c928-4ab5-861b-48f6ff395c67.png)

### Emulation Settings

![imagen](https://user-images.githubusercontent.com/5641681/208320665-128aee58-a1e4-4564-9a8e-75c551cb7558.png)

**Comment by legend800** [comment]
I can confirm this - you cannot bypass the cali screen on Vampire Night because of this. 

Time Crisis Crisis Zone also suffers from the same issue, except 1 out of every 20-30 shots I can get it to work on cali screen in that title, never on Vampire Night.

Fwiw, both of these games previously required nuvee plugin's setting "mouse calibration hack" and work perfectly with it on:
![2022-12-18 14_32_17-Guncon - Port 1](https://user-images.githubusercontent.com/20983844/208332362-b080c8fd-6ef4-4f39-9b14-4a532aea90fa.png)

Related, this can be closed now - it was locked and I can't do it:
https://github.com/PCSX2/pcsx2/issues/3931

**Comment by IlDucci** [comment]
> Time Crisis Crisis Zone

I was able to properly calibrate and take the proper measures from the PAL version with no problem.

**Comment by Comfarol** [comment]
try this for pal vampire night 
![image](https://user-images.githubusercontent.com/27315987/208422510-c440ab00-0927-4832-96fa-5d476015335d.png)

You need to find the correct values

**Comment by IlDucci** [comment]
> try this for pal vampire night

Welp, those values actually unblock the calibration screen issue. The scale seems to be a teeny bit off when the cursor is close to the top and bottom, but at least the game works.

**Comment by Comfarol** [comment]
I can't test alot of the other games as I don't own them but try, see if those are better, looks better for me.
 
89.80
99.90
421.00
138.00
640px
256px

**Comment by dante3732** [comment]
This works well for vampire night usa. But as we wrote above, you need to take several shots to calibrate.
![Vampire Night USA](https://user-images.githubusercontent.com/66036249/208622288-3893966f-618b-48f5-9757-dc05ef2625cb.png)

**Comment by Comfarol** [comment]
Vampire Night USA is 640x448

Are you using the calibration button you bound in the settings, it calibrates every shot for me.

I used yours but fine tuned it on my end and using

89.80
105.10
422
124
640
224

These values any good for you, does it improve your calibration issue?

**Comment by dante3732** [comment]
> Vampire Night USA is 640x448
Yep, you right.

> These values any good for you, does it improve your calibration issue?
Yep, you value fine, but my value little better for me.

**Comment by Comfarol** [comment]
Someone will have to update it with the above values then, by default its loading those cooked values

**Comment by IlDucci** [comment]
> Someone will have to update it with the above values then,

I can do that and add the other values I figured out.

**Comment by dante3732** [comment]
> I can do that and add the other values I figured out. The thread is still valid until the issue with Dino Stalker is fixed.

Dino Stalker PAL? USA work fine for me.

**Comment by IlDucci** [comment]
> Dino Stalker PAL?

PAL, yes. See the first message.

**Comment by Comfarol** [comment]
Dino Stalker PAL has no default values

**Comment by dante3732** [comment]
I test 11 games.
Time Crisis jap OK
Time Crisis 2 usa OK
Time Crisis 3 usa OK
Crisis Zone usa OK, (If after calibration, the rifle can only fire single shots, in which case you need to repeat the calibration again until the rifle starts shooting properly.)
Dino Stalker usa OK
Resident Evil Dead usa OK
Gunfighter II europe OK
Endgame usa OK, but need plug guncon in usb 2

Virtua Cop pal BAD calibration work good, but there is a very long delay before the shot is fired
Vampire Night usa BAD can't calibrate
Resident Evil Gun Survivor 2 pal BAD the mouse cursor does not match the sight in calibration

**Comment by Comfarol** [comment]
Have you  tried VC since update? It had bad values.

**Comment by dante3732** [comment]
> Have you tried VC since update? It had bad values.

Yep, i tested again now. Calibration good, but input lag too big, approximately 2 seconds.

**Comment by Comfarol** [comment]
I'm not getting any lag when firing on my end.

**Comment by dante3732** [comment]
> I'm not getting any lag when firing on my end.

After recalibrating in the main menu, there was no longer such a delay. 
Apparently there is still some kind of calibration problem.

**Comment by IlDucci** [comment]
I just realized that Dead Aim's settings were targeted to the PAL version, not the NTSC-U one. Anybody has Dead Aim on NTSC-U?

**Comment by dante3732** [comment]
> I just realized that Dead Aim's settings were targeted to the PAL version, not the NTSC-U one. Anybody has Dead Aim on NTSC-U?
Yep, USA work fine for me, but i use no interlacing patch.

**Comment by Comfarol** [comment]
You doing Vampire Night USA values from above too?

**Comment by IlDucci** [comment]
Nope, those were the ones from before. I wasn't sure which one of the two sets were the final ones. Can you tell me?

**Comment by Comfarol** [comment]
89.80
105.10
422
124
640
224

**Comment by Comfarol** [comment]
The other games you guys are talking about i don't own so i cant help you there

**Comment by IlDucci** [comment]
> My value just points to the center of the circle in each corner, and yours shifts a little, but it not critical.

What about that?

**Comment by Comfarol** [comment]
I dunno its perfect to me. he seems to be having issues so im biased :D

**Comment by dante3732** [comment]
> Please test it with the values that I just uploaded here: https://github.com/PCSX2/pcsx2/pull/7636/files .
> And also make sure you have no patches enabled.

Looks fine.

**Comment by IlDucci** [comment]
> I dunno its perfect to me. he seems to be having issues so im biased :D Either is fine you talking a few pixels

Try this (and let's hope GitHub doesn't pose any trouble): do a little video recording of yourself on the calibration screen, with the mouse cursor being seen (OBS Studio should be best for this), after doing the calibration, but before leaving the calibration screen. Simply move the cursor to the center of the screen and then to the four corners, as close as you can to the screen limits.

**Comment by Comfarol** [comment]
its like two pixels between his/mine. im not making a video sorry, use his

**Comment by IlDucci** [comment]
> its like two pixels between his/mine. im not making a video sorry, use his

Done.

**Comment by Comfarol** [comment]
Dino Stalker PAL works with values, its a little bit off on the edges if you want to fine tune.

![image](https://user-images.githubusercontent.com/27315987/208670196-c866dddf-835e-4373-a913-0efa14d3d436.png)

**Comment by dante3732** [comment]
I captured input lag in veronica and virtua cop. The trigger is pulled when the screen flashes. And the shot happens in 2 seconds.
https://drive.google.com/file/d/1XWx50MKMA3WV3JXPaPknlxMXqnXAs6wv/view?usp=share_link
https://drive.google.com/file/d/1vp2YtPC-j-VLjiLXy401bPYW-gAeNjN-/view?usp=share_link

**Comment by refractionpcsx2** [comment]
I don't think that's input lag, the input triggers the white screen, not the visual shot, the game does that.

**Comment by legend800** [comment]
I only get the long black screen if I go way out the circle on cali (with default values). So there's still something funky here but it's not really impacting. I see you can get a lot of shots off fine when on the target. 

Are you sure you calibration is saving? As I mentioned in the other thread, the steps are convoluted, but it works perfect for me if I do them. 

Press Start, fire in center, fire offscreen, select Ok, with Press Start up, fire offscreen again. If you do that, you'll get this:
![2022-12-20 09_37_15-Virtua Cop - Elite Edition](https://user-images.githubusercontent.com/20983844/208731408-1e3fe6da-b0f9-4e07-99ba-a7f934ec211d.png)
Select Yes and go in-game

**Comment by dante3732** [comment]
> Press Start, fire in center, fire offscreen, select Ok, with Press Start up, fire offscreen again. If you do that, you'll get this:
Yep, i make same.

**Comment by Tovarichtch** [comment]
Do you use the internal default resolution for manual screen calibration? For example : `SLUS-20927 Time Crisis - Crisis Zone`. It works by default, thanks to precalibration made, but it's not enough perfect for my taste. I've done mine, but...

![calibration](https://user-images.githubusercontent.com/32983607/208737721-0a1b2f87-2099-423b-8ebf-a9b060243a73.png)

The values I got if I use the internal resolution are different from the ones merged because of the simulated screen values. Am I doing this wrong?

**Comment by Comfarol** [comment]
https://user-images.githubusercontent.com/27315987/208787578-1a6bb445-9e6a-45bd-a4ba-848aefa710bc.mp4

Don't press start. It seems if you press start then try calibrate you get the black screens and delays. If you just Calibrate and shoot off screen and pick yes its fine.

**Comment by Comfarol** [comment]
Mine isn't modded and I have no cheats. Only graphics setting i have is high blending and anti blur.  Maybe yours is modded and has cheats?

When i start the game the cross hair is perfectly positioned with out calibrating, maybe that's why. What if you use custom settings to adjust it into position and not press start like what i do.

**Comment by legend800** [comment]
I'm sure that'll work too. If the "auto" defaults line up my gun cursor with mouse cursor, then yeah, I can just skip all this stuff. That's ideal. But until the default does that, it's easier for me to go through the full cali steps vs. changing custom calibration settings, since I'm already using current values for Vampire Night. :)

**Comment by Comfarol** [comment]
Why do you need settings for Vampire Night, do the defaults also not work for you on that game too?

**Comment by legend800** [comment]
Yeah, there was a bunch of conversation on this in the middle of the thread above. And only works with custom settings right now, until a new PR is cut.

**Comment by Comfarol** [comment]
Pretty sure its out? Mines working now with no custom settings.

**Comment by Comfarol** [comment]
Anyway with VC it seems the default values give different results for others for some reason? A per game GunCon 2 custom setting option would be the ideal way to get it accurate for individuals (and other games) until what ever is scuffing the calibration screen is found/fixed. It is scuffing because the JPN version calibrates with the same black screen but not prolonged blackness with delayed shooting.

**Comment by legend800** [comment]
> Pretty sure its out? Mines working now with no custom settings.

Ah, nice. Yep, Vampire Night (usa) is fixed now using automatic/default settings on the latest nightly.

**Comment by dante3732** [comment]
By the way, I was wrong about first time crisis. Disc have name gunvari collection + time crisis. There are 2 calibrations, the first one goes fine at startup, and the second one does not when you select time crisis in the menu.

**Comment by shanethmoore** [comment]
I've noticed a lot of comments saying that Time Crisis 2 and 3 are perfect but I'm not so sure thats true. For single player calibration and gameplay it works fine but with split screen the calibration seems to be ignored and instead the cursor shots shows float outwards on the X and Y axis from p1 center screen or p2 center screen.

Unsure if it's calibration related or something else entirely. Could be worth mentioning here or https://github.com/PCSX2/pcsx2/issues/7619 

I'd wager https://github.com/Comfarol/pcsx2/blob/e018c1ac06ed49fb3cc68be66f3f55f4ec67ce65/pcsx2/USB/usb-lightgun/guncon2.cpp#L79 needs to be updated based on whether the game is single or multi player.

**Comment by legend800** [comment]
Good point. 2p was mentioned in the original PR that it would be added/supported in a feature commit. 

FWIW, here's the values I used in nuvee plugin for either SP or MP for reference:

;time_crisis2 "Time Crisis 2 (U) - 2p"
95 45 390 150
namco 640 240

;time_crisis2 "Time Crisis 2 (U) - 1p"
120 97.5 390 154
namco 640 240

;time_crisis3 "Time Crisis 3 (U) - 2p"
95 45 390 150
namco 640 240

;time_crisis3 "Time Crisis 3 (U) - 1p"
120 97.5 390 154
namco 640 240

**Comment by shanethmoore** [comment]
@legend800 Thanks for sharing. I missed that comment about 2p in the original PR

Unfortunately I wasn't able to get 2p with the same accuracy as the 1p. I tried on NTSC and PAL with a lot of different values including the ones you provided. I always seemed to just be bit of the mark or seeing completely backwards behaviors.

If you ever manage to get a configuration working with 1.7+ on your machine, please do let me know 🙏 🙇

**Comment by dante3732** [comment]
It think crisis zone have regression on last qt. Shooting works fine in the menu, but in the game itself, not a single bullet hits the screen. I downgrade to 3807 and everything is fine there.

**Comment by refractionpcsx2** [comment]
can you try again on  3817 when it builds, there was a regression from 3813 which has now been fixed

**Comment by dante3732** [comment]
Okay, I'll try.

**Comment by dante3732** [comment]
Yep, on 3817 it work fine. It seems that calibration has now started to work better in this game.

**Comment by refractionpcsx2** [comment]
Thanks for checking

**Comment by dante3732** [comment]
Crisis Zone, Virtua Cop and Resident Evil Survivor 2 still have calibration issues.

Crisis Zone can only shoot single shots, but problems can be avoided if you hold down the calibration button and hold it in the center for a couple of seconds.

In Virtua Cop, there may be a long delay after the shot, but problems can be avoided if you calibrate until the screen is just blinking instead of remaining black.

If progressive mode is activated in these games and 2 calibration points appear, then the above problems cannot be workbound.

In Resident Evil Survivor, it is almost impossible to calibrate the sight. This can be done on the paintwork, but there will be a very long delay of the shot, as in virtua cop.

If necessary, I can record a video of these problems.

**Comment by legend800** [comment]
Wow, I can't believe TC2, TC3, TC CZ and Vampire Night all have Progressive scan mode! They're not listed on various "progressive scan' game lists, so glad you mentioned that. Now these games are even better!

I still have zero missing/laggy input with TC CZ and Virtua Cop, progressive mode or not, without having to do any workarounds like you mentioned. I simply boot the games, press the cali button once on the target in cali screen, exit and in-game things are aligned with no missing input. I've tested all these games tens of times and never experienced this once with my mouse. 

I just want the devs to know this is only impacting you so far. I have not seen reports from others on TCCZ or Virtua Cop. Clearly the videos show this strange delay/missing input is happening for you, but it will help with root causing to know it's fine for others.

Have you tried a full, clean reinstall of the app? Do you have some nvidia panel low latency setting on? Some Windows setting like Game Mode on? Clearly something input wise is off with your config, else it would not be perfect with my config using the same emulator.

**Comment by dante3732** [comment]
> I just want the devs to know this is only impacting you so far. I have not seen reports from others on TCCZ or Virtua Cop. Clearly the videos show this strange delay/missing input is happening for you, but it will help with root causing to know it's fine for others.

Well, then we will assume that the problem is related to AMD.
I have tested 11 games, 8 are working fine. In 2, there is a calibration problem that can be circumvented. And 1 RE Survivor 2 is almost impossible to calibrate.

**Comment by SeongGino** [comment]
Only just got around to seeing Guncon support added, so may as well throw my input in (Linux, 1.7.3871 rn):

TC2/TC3/Crisis Zone/Vampire Night/Ninja Assault (all NTSC-U) work just fine ootb. 
Virtua Cop ReBirth (NTSC-J) works but default cursor was off-centered and aiming seemed over-sensitive, so I set these offsets manually in its own dedicated control profile to work with the game's default calibration (see attached:)
![2023_01-07-204616](https://user-images.githubusercontent.com/7321839/211177046-7e31df82-92e9-41df-b23f-6ac27dd45c1c.png)
Death Crimson OX+ (NTSC-J) works and I can adjust the X-axis center and sensitivity, but the Y-axis seems perpetually over-sensitive no matter how much I change the setting. I'm not sure what's up with this one.
Biohazard Gun Survivor 2 (NTSC-J) doesn't calibrate well at all, the cursor can't seem to get far past the upper-fourth of the screen and is always going to be offset somewhere to the left and up of center.
Need to test Dead Aim and Dino Stalker eventually but I don't have those atm.

If it matters, Arch Linux kernel 6.1.2, NVIDIA 525.78.01, Plasma X11.

**Comment by Tovarichtch** [comment]
I concur with @legend800 : Virtua Cop (and few other games) are working fine on the calibration screen and afterwards _if you use the calibration button_ and not just shooting on the screen.

**Comment by retrofan2019** [comment]
first of all amazing work by the devs .. pretty easy and straight forward to set up ...
I tested most of the gun games and here is my feedback hopefully that might help :

as stated by other users all the US titles r working perfectly fine 

-Time Crisis II
-Time Crisis 3
-Crisis zone  ( as pointed out by @dante3732  need to hold the "calibration shot" button at the center for a few seconds ... it took me a few times to finally get it working as intended )
-Vampire Night
-Ninja Assault
-Dino Stalker  (needs "software renderer FMV hack" on to get rid of the fmv video problem)
-Resident Evil : Dead Aim  

as for European titles I tested 4 games:
-Gunfighter II: Revenge of Jesse James - works perfectly fine 
-Virtua Cop: Elite Edition ( works perfectly fine as long as the calibration is done right ... as pointed out by  @dante3732 .. press the "calibration shot" button if the screen flashes the back screen and returns to calibration than it's all good .. shoot outside the screen to get the menu press "ok" than shoot again offscreen to get out of the calibration menu ... if you get a black screen after the pressing the "calibration shot" button ..than u have to keep trying to get it right )
- Guncom 2 (aka Death Crimson OX)  - took a few times to get the aim perfect in the calibration screen but once u get it right it works perfectly fine in game.
- Resident Evil Survivor 2 Code: Veronica - I see that most people failed to make it work .. but it actually worked ok for me .. took me a few tries to get the aim good in the calibration screen .. the best way to do so is use "calibration shot" when u get the center target than use the trigger for the top left target ...once it's working fine in the calib screen .. it works well in the game ..not perfect aim but it's decent ..it's slightly off in some parts of the screen .

Japanese titles :

-Gunvari Collection + Time Crisis : I only tested time crisis and as stated above you need to calibrate for menu and than calibrate for game .. and it worked fine ..like RE :Survivor the aim was not perfect but it's playable.

Death Crimson OX and Virtua Cop - both games don't work really well ,the aim is off specifically in the left side of the screen .

finally 2 quick notes :
- the "calibration shot" button is not the trigger button ..and it's the main button used to calibrate all the gun games .
- for people having trouble calibrating certain games .. and it takes a few tries to get it right ..save state can be used after a correct calibration to save time when playing the game again .

**Comment by shanethmoore** [comment]
@retrofan2019 Did you test Time Crisis 2/3 in split screen?

**Comment by refractionpcsx2** [comment]
If you're testing split screen, I would recommend using "Screen offsets" option, otherwise the picture will be weirdly stretched out

**Comment by refractionpcsx2** [comment]
the boxes are small because you're displaying a 4:3 image in a 16:9 window :P so it doesn't really take up all the space, they did that with the game to maintain the aspect ratio in 2 windows

**Comment by legend800** [comment]
I understand that, I was just trying to see if there's any way to utilize all that black space in the screenshot above

**Comment by refractionpcsx2** [comment]
you could try using the "Crop" functionality, if you're lucky :D

**Comment by retrofan2019** [comment]
> @retrofan2019 Did you test Time Crisis 2/3 in split screen?

I just tested it .. and with screen offset checked as @refractionpcsx2 recommended .. the aim was slightly off on the sides ..but it's not that bad , I would say it's playable ...but definitely not perfect aim for me like when i set it with just one player.

**Comment by SeongGino** [comment]
Wait, so do multiple mouse devices work? I'm assuming that's mostly tested by Windows users?

I unfortunately don't have two devices (or even a way of making my gun device work atm) to check this on Linux, and wasn't sure.

**Comment by retrofan2019** [comment]
> Wait, so do multiple mouse devices work? I'm assuming that's mostly tested by Windows users?
> 
> I unfortunately don't have two devices (or even a way of making my gun device work atm) to check this on Linux, and wasn't sure.

well, actually I only used a mouse (which of course means one gun will work perfectly fine) ...I just tested it on usb port 1 than moved it to port 2 to test calib for the 2 players .. and I would say that it seems that results on my setup are similar to @dante3732 (with the exception of the resident evil game)  the aim is slightly off in split screen and not perfect like one player ...didn't try manual screen configurations .

**Comment by dante3732** [comment]
> with the exception of the resident evil game

And you not have big input lag in gun survivor code veronica?

**Comment by retrofan2019** [comment]
> 

there is a slight input lag ... but the aim is actually not that terrible ...just took me a few tries to get it to work but once it's fine in the calib screen it works well in game ... I would say the aim is better than dual screen time crisis .
overall I would say it's playable .. but i only played it for like 15 mins or so cuz it's not really a good game .

**Comment by dante3732** [comment]
> there is a slight input lag ... but the aim is actually not that terrible ...just took me a few tries to get it to work but once it's fine in the calib screen it works well in game ... I would say the aim is better than dual screen time crisis .
> overall I would say it's playable .. but i only played it for like 15 mins or so cuz it's not really a good game .

Better than in my video? https://drive.google.com/file/d/1XWx50MKMA3WV3JXPaPknlxMXqnXAs6wv/view

**Comment by shanethmoore** [comment]
@SeongGino For testing split screen I used a single mouse and bound it to p1 and p2. Left clicking fires both players guns but only shoots depending on which side of the screen you're pointing at.

> I just tested it .. and with screen offset checked as @refractionpcsx2 recommended .. the aim was slightly off on the sides ..but it's not that bad , I would say it's playable ...but definitely not perfect aim for me like when i set it with just one player.

@retrofan I tested TC2 myself with the same settings and I think my experience was a lot worse than yours. Playable but very frustrating. Theres a much more pronounced drift on both the X and Y from where the mouse points to where the gun shoots. After many many attempts to change the [screen offsets](https://github.com/Comfarol/pcsx2/blob/e018c1ac06ed49fb3cc68be66f3f55f4ec67ce65/pcsx2/USB/usb-lightgun/guncon2.cpp#L79) I still couldn't get it perfect and the values were worthless after recalibrating. 



I know TC2/TC3 are very popular Guncon games, but this thread could runaway trying to solve problems with single games and get locked like the last one.

I think it would be worth putting together a wiki page to track compatibility of Guncon games, some sort of playability score and if any patches/steps are required. 

Thoughts?

**Comment by retrofan2019** [comment]
> > there is a slight input lag ... but the aim is actually not that terrible ...just took me a few tries to get it to work but once it's fine in the calib screen it works well in game ... I would say the aim is better than dual screen time crisis .
> > overall I would say it's playable .. but i only played it for like 15 mins or so cuz it's not really a good game .
> 
> Better than in my video? https://drive.google.com/file/d/1XWx50MKMA3WV3JXPaPknlxMXqnXAs6wv/view

Oh i thought you had problems with the aim ... it seems like the aim is almost perfect for u ... as for input lag are you spamming the shoot button and this is the fastest you can shoot ? if so yea the input lag for you is way worse ...for me yes there is a noticeable input lag but for a really slow and clunky game it's not bad to the point that it renders the game unplayable ..

**Comment by retrofan2019** [comment]
@shanethmoore 

are you talking about TC2 in general or the split screen .. because yea the split screen thing can be frustrating ..the aim is definitely off ... I guess the fact that the screen is tiny is what makes it playable .

and the wiki idea is great ... it can save a lot of time for anyone who is settings things up for the first time and we can send the link to anyone with a question that is already answered .

**Comment by Ikhlasraja** [comment]
PLZXX SEND ME CD FILE COMMDS I NEED CD COMMDS

**Comment by dante3732** [comment]
@retrofan2019 By the way, do you have an AMD graphics card and processor? If you have similar calibration problems in two games, maybe that's the point.

**Comment by shanethmoore** [comment]
@retrofan2019 Specifically split screen. The SP works perfect as you say, but yeah split screen is borderline unplayable for me.

Theres two wikis, the Github which is restricted and the pcsx2.net wiki but signup is broken

<img width="364" alt="Screenshot 2023-01-17 at 11 26 53" src="https://user-images.githubusercontent.com/3028279/212887758-1347548f-3718-4259-8fae-fbd723fda693.png">

I'll check again in a bit and see if we can get something started 👍

**Comment by retrofan2019** [comment]
> @retrofan2019 By the way, do you have an AMD graphics card and processor? If you have similar calibration problems in two games, maybe that's the point.

nope I have a Nvidia GTX 970 and the cpu is i7-4790k.

**Comment by retrofan2019** [comment]
@shanethmoore  yep ,the aim in split screen is just not that good .. idk whats causing it maybe the screen offsets option just doesn't position the screen correctly ..

and yep that would be great  ... I'm sure a lot of ppl will be happy to find a guncon2 compatibility page on the pscx2 wiki .

**Comment by Mrlinkwii** [comment]
what the status of the guncon2 games , is their any games left with issues ?

**Comment by shanethmoore** [comment]
@Mrlinkwii Can confirm TC2 and TC3 2P split screen still has aim problems. I haven't tested any other guncon games in a while and was waiting to see the outcome of https://github.com/PCSX2/pcsx2/issues/7888 and https://github.com/PCSX2/pcsx2/pull/8437

On another note, It seems the pcsx2 wiki signup is fixed, I'd be happy to add it to the list of known issues. https://wiki.pcsx2.net/index.php/Time_Crisis_II

Might be a better way of tracking it than a spreadsheet 😓

**Comment by dante3732** [comment]
> I have tested 11 games, 8 are working fine. In 2, there is a calibration problem that can be circumvented. And 1 RE Survivor 2 is almost impossible to calibrate.

Seems Resident Evil Survivor 2 and Virtua Cop calibration now work fine.

There may still be a problem with single shots in crisis zone because of this.
![pcsx2-qt_2023_08_23_14_10_39_689](https://github.com/PCSX2/pcsx2/assets/66036249/c533e88b-b2f8-441f-b5f1-29ddadeb7c09)

**Comment by ZTylerDurden** [comment]
Since there has been no mention yet of Sinden/Gun4IR light gun solutions mentioned here, could I take a step back and confirm what's going on here?

My understanding is that 1.6 using nuvee "patched/hacked" these calibration issues per game basis.

1.7 with built-in usb Guncon2 emulation is having growing pains with calibration issues per game basis.

So regardless of whether Sinden/Gun4IR has their **own separate layer of calibration at the OS level**, calibration in-game is a sub-layer of calibration that could make or break an accurate aiming experience?

**Comment by shanethmoore** [comment]
@ZTylerDurden ✅ 

I'll try elaborate my understanding and sorry if any of this is mentioned elsewhere, I'll try to be thorough. I have 3 Sinden guns and 1 Gun4IR. The solutions they provide only go as far as ensuring that where the gun points at the screen can be translated to accurate mouse coordinates on the machine (OS level calibration as you said). 

You're right in saying PCSX2's per game calibration is where things start to become complicated. 

You can actually just use a mouse or 2 as a simulated PCSX2 guncon to replicate the problem. We have the ability to set per game values for where the mouse pointer lands, and where the gun shoots. Those values need to be fine tweaked per game, per screen, size per aspect and I suspect per region (not confirmed).



The best example I can think of where this falls apart is in Time Crisis 2 on a bog standard screen at native resolution on a 1920 x 1080p monitor. If I set the values to be the [default as in the code](https://github.com/Comfarol/pcsx2/blob/e018c1ac06ed49fb3cc68be66f3f55f4ec67ce65/pcsx2/USB/usb-lightgun/guncon2.cpp#L80):

```
time_crisis2 "Time Crisis 2 (U)"
90.25 97.5 400 118
namco 640 120`
```

1. I _might_ get a playable 1p game for the most part with some slight offset issues in some areas. The only way around this is to continually tweak the values in game until I find the sweetspot of playability, but I have never gotten it 100% accurate from mouse -> in game gun


2. If I go into split screen 2p, those same values won't work and the game is unplayable once the screen *splits*. 

As a workaround, I can try to setup the values to work for 2p using different values for each device. The aiming is more noticably off but it becomes playable, to get a playable game (although the aiming is still a lot more noticably off)

But then you create a new problem in that the coordinates you're using work for 2p split screen, but not for the main menu or the calibration screen.

It involves juggling a fair amount of save states to get that far.


3. Lastly, I've not confirmed in TC2 but I've seen some games have calibration screens where they require you to use a different button than the normal trigger. Thats fine, but after the calibration the games aspect ratio or values seem to change. 

So you end up needing to calibrate with one set of offsets, saving a state after calibration and then switching to a playable set. Same as #2 👆 



Tl;DR per game calibration is a lot of hoop jumping per instance to get it right and even then, it can only get so far.

**Comment by ZTylerDurden** [comment]
@shanethmoore That's as clear as it gets.  Really appreciate that.  At least now anyone who searches keyword "Sinden/Gun4IR" can be redirected here to understand what's going on.

Would you mind sharing your manual coordinates for split screen TC2?  So basically you're calibrating each player at boot, saving state after calibration, then manually overriding each P1/P2 guncons coordinates, then loading state back in game? 1080p widescreen overscan/offsets disabled? I can use that, even if not pixel perfect.

**Comment by legend800** [comment]
I think you guys are over-complicating this a little. Some info from my experience playing guncon games on pcsx2 for many years (I can elaborate as needed).

- "Automatic Parameters" works fine for all my games including TC2 (after calibrating in-game). Perfect in-game cursor > shots.
- You can override if needed but the sensitivity, centering and "simulated screen" w/h should be fine for everyone out of the box. 
- Everything worked perfectly with nuvee until they removed plugin support
- Nuvee is the ONLY way to do 2 player (with guns/mice) since it supported Rawinput and pcsx2 does not (yet)

**Comment by ZTylerDurden** [comment]
@legend800 

Thanks. 

Your first bullet: PCSX2 1.6 or 1.7? Split-screen 2 player included?

**Comment by legend800** [comment]
pcsx2-v1.7.0-dev-549 was the last working build for nuvee, so forget 1.6. :) Haven't tried split screen since plugin removal since P2 can only be gamepad. 

Lichtknarre? Haven't heard of it, but I'll go play around with it now and update my results here if I can get it running.

**Comment by ZTylerDurden** [comment]
Excellent.

Just to confirm the plug values you provided above

**TC2 Split-screen mode**
**P2**
95= X scale (sensitivity) 
45= Y scale (sensitivity)
390= Center X
150= Center Y
640= Screen Width
240= Screen Height

etc. for P1 values.

**Comment by ZTylerDurden** [comment]
Excellent.

Those manual values are compatible with split-screen TC2 or is that a different set?

**Comment by legend800** [comment]
This is just what I used in nuvee for 2p split screen. I think it worked fine but been a few years. I'm just hoping they had some kind of support for 2p/split-screen in games like this.

time_crisis2 "Time Crisis 2 (U) - 2p"
95 45 390 150
;namco 640 240

**Comment by ZTylerDurden** [comment]
Understood now.  Thanks again.

**Comment by IlDucci** [comment]
> You can actually just use a mouse or 2 as a simulated PCSX2 guncon to replicate the problem. We have the ability to set per game values for where the mouse pointer lands, and where the gun shoots. Those values need to be fine tweaked per game, per screen, size per aspect and I suspect per region (not confirmed).

Resolutions and regions go hand in hand, so yeah, per resolution/region (some PAL games have identical values to their NTSC-U counterparts, some don't, though I do not know if those are 100% correct or if there's more than one setting per copy of game that actually worksTM).

Local values can be found here: https://github.com/PCSX2/pcsx2/blob/56ec842c008ed3cc0c43b3ba12186359c3b03eb0/pcsx2/USB/usb-lightgun/guncon2.cpp#L76

**Comment by stenzek** [comment]
> Nuvee is the ONLY way to do 2 player (with guns/mice) since it supported Rawinput and pcsx2 does not (yet). That's the ONLY benefit to using an older build of pcsx2

Wrong. You can do two lighthouse if you use relative/axis mode instead of mouse mode.

There isn't a clean cross-platform way to do multiple mice, even in windows with raw input it's super jank. Plus users being utter twats about it (see https://github.com/PCSX2/pcsx2/pull/8437)

> Everything worked perfectly with Nuvee until they removed plugin support

Comments like this are frankly disrespectful to the hundreds of hours that many of us have collectively put into PCSX2 1.7/2.0. If you want to use ancient stuff, sure, go nuts, please keep such comments to yourself - it just further reduces anyone's motivation to work on these features.

**Comment by legend800** [comment]
> Wrong. You can do two lightguns if you use relative/axis mode instead of mouse mode.
I know about relative mode but I wasn't counting things that don't work well for rail shooters. This thread is about GunCons also. Have you used a lightgun, in relative mode and play Time Crisis? Yeah, it's rough. Usable? Yes, but not proper input for a quick moving rail shooters where you have to jump from one corner of the screen to the another in a split second. 

I've had zero issues with raw input over the years. Not sure what makes it "jank". It works perfectly in Mame, Supermodel 2 and 3 emus, Demulshooter, Teknoparrot and many others to using multi-mice/lightguns.  It's the only way to have a proper 2 lightgun experience.

> Comments like this are frankly disrespectful to the hundreds of hours that many of us have collectively put into PCSX2 1.7/2.0. If you want to use ancient stuff, sure, go nuts, please keep such comments to yourself - it just further reduces anyone's motivation to work on these features.

Easy. That had nothing to do with the decision or effort to drop plugins. You're taking that comment waaaay wrong. Just stating a fact. I could use 1p and 2p lightguns perfectly before, when we had rawinput support via Nuvee. Now we don't, so it's been a new limitation since Aug 2020. I'm glad they got the implementation this far and hoping they can find a way to support 2p properly. Nuvee also spent hundreds of hours on his plugin to just see thrown out too. :)

**Comment by stenzek** [comment]
> Have you used a lightgun, in relative mode and play Time Crisis? 

No, I don't even own a lightgun. Which is what makes your complaining even more silly - I'm the one who went out of my way to implement this support, actually rewriting the entire USB subsystem, because nobody else wanted to or cared. I waited almost a year for the previous maintainer to bring USB over to Qt, and they couldn't even be bothered to respond to our questions.

But the feedback when I did implement relative mode was positive, unlike anything you're saying.

> I've had zero issues with raw input over the years. Not sure what makes it "jank". 

I already told you. It's not cross platform, at PCSX2 we don't care only about Windows. It also conflicts with regular window mouse messages.

> Nuvee also spent hundreds of hours on his plugin to just see it thrown out too. :)

Do you see them around here? Seems they don't care either. There's a reason that **basically no** emus use plugins these days. Something which "worked perfectly", considering it didn't even know what game was running, because the plugin interface didn't provide it, and couldn't automatically choose a profile? Doesn't sound very "perfect" to me.

> Now we don't, so it's been a new limitation since Aug 2020. I'm glad they got the implementation this far and hoping they can find a way to support 2p properly.

So? You're not our boss. We're all volunteers, and don't owe you anything. If it's so important, then feel free to implement it yourself. Furthermore, it's not "they". It's literally just me. Comments like this basically make me say "well, why should I do anything for these ungrateful assholes, when I'm not personally invested in the feature? I'm gonna work on other fun things instead".

**Comment by legend800** [comment]
1) Wow...
2) No one is complaining.  We're stating information here about the state of lightgun emulation....there's a difference. 
3) Relative mode does not work well for Rail shooters. It's usable, that's all, but it's not the best way to play to rail shooters.
4) Fine, you're right, Nuvee was not perfect because it required me to select a game profile before playing. You win.
5) How does stating there's a new limitation and saying **_I hope_** we can get full 2p lightgun support...make it sound like I'm trying to be a boss? That's funny. Where do I say you owe me anything? It's like you are looking for anything to say it's an attack or a complaint. We're stating facts here. Calm down. You are taking every statement like an insult for some reason. Not productive. I'm an ungrateful asshole now? What?? Read what I wrote a few more times. It couldn't be more neutral. Run it through ChatGPT and see what he thinks. You are being incredibly sensitive and not sure why you're picking fights here when I was just helping. All because I said dual lightguns worked perfectly before plugin support was removed?? Now I know why you quit the PR, because any comment, **_ANY_** comment is apparently an insult to you and the other person is just being an asshole.

**Comment by stenzek** [comment]
And given this is turning into flames, I think it's time we ended the discussion. Just like the PR thread, for some reason this topic always seems to end up going in this direction.

**Comment by shanethmoore** [comment]
Jeepers, I wasn't expecting this level of response when I left my laptop this afternoon. I'd rather not see this thread locked like the [last one](https://github.com/PCSX2/pcsx2/pull/8437) due to any disrespect or going off topic.

@legend800 I would delete your latest comment and leave it at that before this escalates any further. Try have a little empathy for the pressure that Stenzek is under.

@stenzek For whatever my two cents is worth, I don't think any disrespect was intended on legend800's part originally and I don't think he was trying to act like your boss. We very much appreciate all the work thats gone into this project, there really is no equal out there!


One of the few ways us muggles can contribute is by pointing out bugs or feature mismatches, so please don't take anything said here as us being ungrateful.


> Wrong. You can do two lightguns if you use relative/axis mode instead of mouse mode.

I've not tried this method yet but I'd be happy to give it a shot or test out any other suggestions listed today.
